user-script

```diff
@@ -31,39 +32,61 @@ fatal() {
 msg "called $(date)"
 
 # Get the username, home directory, and DISPLAY number of the currently logged
-# in user with an X session.
+# in user with a Wayland or X session.
 #
 # This logic is fairly dumb: it will stop at the first display found, and give
 # up easily if anything is awry.
 #
 # This should probably be extended to support multiply displays and multiple
 # users... but for now this gets the job done for a single user workstation.
 found=false
-for sock in /tmp/.X11-unix/X*; do
-	# socket group is the user that owns the session
-	user=$(stat --printf '%G' "$sock")
-	[[ -n $user ]] || fatal "failed to find owner (group) of $sock"
+# search for wayland session
+for xrd in /run/user/*; do
+	userid=${xrd##*/}
+	user=$(getent passwd "$userid" | cut -f 1 -d ":")
+	[[ -n $user ]] || fatal "failed to find owner of $xrd"
+	for f in "$xrd"/*; do
+		if [[ $f == *"wayland"* ]]; then
+			# extract the display number
+			display=${f##*/}
+			[[ ${display#"wayland-"} =~ $num_re ]] || \
+				continue
+			break
+		fi
+	done
+	if [[ $display == *"wayland"* ]]; then
+		found=true
+	break
+	fi
+done
+# search for X session
+if [ "$found" = false ] ; then
```

Comment:
style nit: since the contents of the `$found` variable are controlled by this script for the duration of its lifetime, you can rewrite this to match how I have it below:

``` bash
if ! $found; then
  ...
fi
```